### PR TITLE
Change CMake config install dir to match one of the standard locations

### PR DESCRIPTION
**Pull request by davidbeckingsale**
This means that `RAJA_DIR` can just point at the install prefix and `find_package` will work correctly. It will no longer require specifying the share/cmake/ subdirectories.

Other paths will work (e.g. share/raja/cmake), as specified here: https://cmake.org/cmake/help/v3.0/command/find_package.html

I'm happy to change it to one of those other paths if people prefer.

### CODE REVIEW

**Comment by DavidPoliakoff** [comment]
This looks good to me.

**Comment by davidbeckingsale** [comment]
It seems like our initial location just had the `raja` and `cmake` the wrong way round, but moving the `cmake` directory to the top-level of the install seems better.

**Comment by DavidPoliakoff** [comment]
This isn't to assign blame, but it's probably whoever wrote [that code's](https://github.com/LLNL/Caliper/blob/master/CMakeLists.txt#L133) fault.

**Comment by davidbeckingsale** [comment]
@DavidPoliakoff can you approve this?

**Comment by davidbeckingsale** [comment]
@trws do you have any thoughts on the best location for the CMake config that get's installed?

**Comment by trws** [comment]
I suppose under share is usual... Come to think of it, it might be nice to include a packageconfig file as well, with the same information in it (at least as much as is feasible).  That actually might even simplify the use of an installed version from other CMake builds (while the cmake file is clearly superior for builds with source).

**Comment by willkill07** [comment]
@trws I added simple pkg-config generation and tried to encapsulate as much information as possible. Vanilla RAJA works (no external projects) and cflags work for both nvcc and clang

If RAJA was configured with CUDA, getting code to compile was as easy as using this Makefile:
```
CXX := nvcc
CXXFLAGS := $(pkg-config --cflags RAJA)
LDFLAGS := $(pkg-config --libs RAJA)
```

A tag for CUDA-enabled vs CPU-only RAJA would be nice, but I'm not sure about the feasibility given that we are generating the pkg-config from CMake.

**Comment by trws** [comment]
Yeah, I don't see a better way than that, I guess the cflags could include a define for raja-enable-cuda or some-such, but I'm not sure that adds much.  Thanks for setting that up @willkill07!

**Comment by davidbeckingsale** [comment]
I have moved some stuff around to try and centralize the config stuff and keep the root `CMakeLists.txt` clean. The cmake stuff is installed in `/share/raja/cmake`, and that structure is also mirrored in the repository.

**Comment by DavidPoliakoff** [comment]
:+1: if it passes CI, this looks really good

**Comment by willkill07** [comment]
👍 on this -- this looks a lot cleaner

**Comment by rhornung67** [comment]
:+1: Much cleaner. Thanks, David.

File: cmake/SetupRajaConfig.cmake [hunk 1]
```diff
@@ -87,3 +87,39 @@ if (RAJA_TIMER STREQUAL "cycle")
...
+
+# Configure CMake config
+configure_file(${PROJECT_SOURCE_DIR}/share/raja/cmake/RAJA-config.cmake.in
+  ${PROJECT_BINARY_DIR}/share/raja/cmake/raja-config.cmake)
+
+install(FILES ${PROJECT_BINARY_DIR}/share/raja/cmake/raja-config.cmake
+  DESTINATION share/raja/cmake/)
+
+# Setup pkg-config
+include(FindPkgConfig QUIET)
```
Comment (trws) [inline cmake/SetupRajaConfig.cmake]: Does this always do the presence check? I'm used to seeing it as `find_package(PkgConfig)`.  Aside from that, this all looks like an improvement.

**Comment by willkill07** [comment]
@trws I honestly don't know. I found an example of incorporating package-config generation with CMake and just used the stencil provided. I can take it for a spin later unless someone beats me to it.

**Comment by davidbeckingsale** [comment]
@willkill07 It seems like find_package(X) and include(FindX) do the same thing, but `find_package` might be preferred: http://public.kitware.com/pipermail/cmake/2008-September/024035.html

**Comment by willkill07** [comment]
@davidbeckingsale that's handy to know! I'd make that change now, but I'm currently not by a computer.

**Comment by davidbeckingsale** [comment]
No worries, I got it.

**Pull request merged**